internal/driver/checkstatuses.go

```diff
@@ -25,8 +25,24 @@ func (d *Driver) checkStatuses() {
 
 		status := d.testConnectionMethods(device)
 
-		if err := d.updateDeviceStatus(device.Name, status); err != nil {
+		if statusChanged, err := d.updateDeviceStatus(device.Name, status); err != nil {
 			d.lc.Warnf("Could not update device status for device %s: %s", device.Name, err.Error())
+		} else if statusChanged && status == UpWithAuth {
+			d.lc.Infof("Device %s is now %s, refreshing the device information.", device.Name, UpWithAuth)
+			go func(device models.Device) {
+				// todo: should we update the device?
```

Comment:
Yes, looks like the Device data changed,  so Core Metadata needs that updated data.
```
device.Protocols[OnvifProtocol][DeviceStatus] = status
```

Reply:
Yeah I'm not so sure what that comment referred to. Each of the sub calls below, and the one above update core-metadata, so i don't think it is that.

I think it might have been a todo written before the code that follows it (ie. should we refresh the device information and network information upon a device coming back online), which has been done.